gatekeeper/server.go

```diff
@@ -702,26 +702,10 @@ func (r *OAuthProxy) newOpenIDClient() (*oidc.Client, oidc.ProviderConfig, *http
 	}
 
 	// step: attempt to retrieve the provider configuration
-	completeCh := make(chan bool)
-	go func() {
-		for {
-			r.log.Info("attempting to retrieve configuration discovery url",
-				zap.String("url", r.config.DiscoveryURL),
-				zap.String("timeout", r.config.OpenIDProviderTimeout.String()))
-			if config, err = oidc.FetchProviderConfig(hc, r.config.DiscoveryURL); err == nil {
-				break // break and complete
-			}
-			r.log.Warn("failed to get provider configuration from discovery", zap.Error(err))
-			time.Sleep(time.Second * 3)
-		}
-		completeCh <- true
-	}()
-	// wait for timeout or successful retrieval
-	select {
-	case <-time.After(r.config.OpenIDProviderTimeout):
```

Comment:
Timeout is triggered by `time.After()` causing `newOpenIDClient()` to return and the anonymous func spawned in it's own go routine on line 706 will leak and poll the provider indefinitely, while also flooding logs.